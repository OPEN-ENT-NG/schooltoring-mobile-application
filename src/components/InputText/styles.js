import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const styles = StyleSheet.create({
  input: {
    height: 50,
    ...Platform.select({
      ios: {
        borderWidth: 1,
        borderColor: COLORS.GREY,
        marginBottom: 15,
        padding: 10,
        borderRadius: 3
      }
    })
  }
});

export default styles;
