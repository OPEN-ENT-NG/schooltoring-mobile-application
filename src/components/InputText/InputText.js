import React, { Component } from "react";
import PropTypes from "prop-types";
import { TextInput, Platform } from "react-native";

import styles from "./styles";

const InputText = props => {
  return (
    <TextInput
      secureTextEntry={props.secureTextEntry || false}
      style={styles.input}
      placeholder={props.placeholder}
      value={props.value}
      onChangeText={val => props.onChangeText(val)}
    />
  );
};

InputText.propTypes = {
  secureTextEntry: PropTypes.bool,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired
};

export default InputText;
