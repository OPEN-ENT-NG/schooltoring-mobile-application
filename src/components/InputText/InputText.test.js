import React from "react";
import { TextInput } from "react-native";
import { shallow } from "enzyme";

import InputText from "./InputText";

const placeholder = "Placeholder text",
  value = "",
  onChangeTextCb = jest.fn(),
  testValue = "value text",
  rendered = shallow(
    <InputText
      value={value}
      placeholder={placeholder}
      onChangeText={onChangeTextCb}
    />
  ),
  input = rendered.find(TextInput);

describe("Render", () => {
  test("InputText should render without crashing", () => {
    expect(rendered).toMatchSnapshot();
  });

  test(`InputText should render placerholder as '${placeholder}'`, () => {
    expect(input.props().placeholder).toEqual(placeholder);
  });

  test(`InputText should render value as '${testValue}'`, () => {
    const component = shallow(
      <InputText value={testValue} onChangeText={onChangeTextCb} />
    );

    expect(component.find(TextInput).props().value).toEqual(testValue);
  });

  test("InputText with secureTextEntry prop display password field", () => {
    const component = shallow(
      <InputText
        value={testValue}
        onChangeText={onChangeTextCb}
        secureTextEntry={true}
      />
    );

    expect(component.find(TextInput).props().secureTextEntry).toBeTruthy();
  });
});

describe("Interaction", () => {
  test("InputText change should trigger onChangeText prop function", () => {
    input.simulate("changeText");
    expect(onChangeTextCb).toHaveBeenCalled();
  });
});
