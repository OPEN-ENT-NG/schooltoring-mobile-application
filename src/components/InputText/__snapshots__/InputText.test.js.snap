// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render InputText should render without crashing 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  placeholder="Placeholder text"
  secureTextEntry={false}
  style={
    Object {
      "borderColor": "#929292",
      "borderRadius": 3,
      "borderWidth": 1,
      "height": 50,
      "marginBottom": 15,
      "padding": 10,
    }
  }
  value=""
/>
`;
